src/butil/containers/hash_tables.h

```diff
@@ -129,8 +129,10 @@ using BUTIL_HASH_NAMESPACE::hash_set;
 //
 // Contact danakj@chromium.org for any questions.
 inline std::size_t HashInts32(uint32_t value1, uint32_t value2) {
-  uint64_t value1_64 = value1;
-  uint64_t hash64 = (value1_64 << 32) | value2;
+  uint64_t short_random1 = 842304669U;
+  uint64_t short_random2 = 619063811U;
+
+  uint64_t hash64 =  short_random1 * value1 + short_random2 * value2;
```

Comment:
please use [fmix64](https://github.com/apache/incubator-brpc/blob/fc07b7bf45c4b2aa90ae5c832f3b48a8da124640/src/butil/third_party/murmurhash3/murmurhash3.h#L57)((value1_64 << 32) | value2) which is better studied.

Reply:
good idea!